### DISCUSSION TITLE
How to prevent Orchard generate CSS and JS with suffix string at the end?

### DISCUSSION THREAD

**Discussion by orchardc** [OrchardCMS/OrchardCore#12358] [Q&A] [answered]
When using the out of box themes, all CSS and JS are generated with a suffix string at the end, for example:

/test1/TheAgencyTheme/css/styles.min.css?v=CTa_kE26-fscUBPBSOFAZEytEoH0v4TVw-g7HVirPOY'
/test1/OrchardCore.Resources/Scripts/popper.min.js?v=1xNSNTNm6oV4Kh94cZjVRB21VYi3qTzka6aeKIWhvMc

Those suffix (?v=xxx) causing problems (**Failed to load resource: the server responded with a status of 404 (Not Found)**) when I try to rewrite a URL, for example:

app.Use(async (context,next) =>
{
    var url = context.Request.Path.Value;
    // Rewrite to index
    if (url.Contains("/home/privacy"))
    {
        // rewrite and continue processing
        context.Request.Path = "/home/index";
    }
    await next();
});

Is there a way to disable those suffix generated at the end of js/css link?

Thanks a lot.

**Comment by orchardc** [comment, marked as answer]
I tried those URLs in the browser, without those suffixes it works just fine.

**Reply by Skrypt** [reply]
This is the assets cache-busting feature from OC. It is useful when you want to publish these on a production server and when you want to bust the client assets cache. I would not remove them.

But here you can disable it there: 

![image](https://user-images.githubusercontent.com/3228637/189179519-58cd4b95-2b9a-4955-a741-14dcf58288c7.png)

**Reply by jtkech** [reply]
You would need to clear the Query/QueryString the part after the Path just after the `?` character. But you have better to do a response redirect and then return to shortcut the middleware pipeline, something like this

                    var pathBase = context.Request.PathBase;
                    if (!pathBase.HasValue)
                    {
                        pathBase = "/";
                    }

                    context.Response.Redirect(pathBase.Add("/home/index"));

                    return;

**Reply by orchardc** [reply]
Hi @Skrypt , indeed it fixed the problem, thanks a lot. :)

Hi @jtkech , Thanks for the tip of QueryString, point taken! I need Rewrite rather than Redirect, mainly because it doesn't change the URL. 

* I can only make the top-level response as  the answer, but the answer credit belongs to @Skrypt
